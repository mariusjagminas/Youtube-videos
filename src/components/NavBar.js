import React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import YouTubeIcon from "@material-ui/icons/YouTube";
import Container from "@material-ui/core/Container";
import { withStyles } from "@material-ui/core/styles";
import SearchBox from "../components/SearchBox";

const styles = theme => ({
  navBar: {
    backgroundColor: theme.palette.grey[100]
  },
  youtube: {
    marginRight: theme.spacing(2),
    fontSize: "30px",
    color: "red"
  },
  title: {
    flexGrow: 1,
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block",
      color: theme.palette.grey[900]
    }
  }
});

const NavBar = ({ onSearchSubmit, classes }) => {
  return (
    <AppBar position="static" elevation={1} className={classes.navBar}>
      <Container>
        <Toolbar variant="regular" className={classes.toolbar}>
          <YouTubeIcon className={classes.youtube} />
          <Typography className={classes.title} variant="h6" noWrap>
            Youtube Videos
          </Typography>
          <SearchBox onSearchSubmit={onSearchSubmit} />
        </Toolbar>
      </Container>
    </AppBar>
  );
};

export default withStyles(styles)(NavBar);
