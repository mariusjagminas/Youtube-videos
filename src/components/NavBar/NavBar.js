import React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import YouTubeIcon from "@material-ui/icons/YouTube";
import Container from "@material-ui/core/Container";
import { withStyles } from "@material-ui/core/styles";
import SearchBox from "";
import LinearProgress from "@material-ui/core/LinearProgress";
import PropTypes from "prop-types";

const styles = theme => ({
  navBar: {
    backgroundColor: theme.palette.grey[100],
    position: "fixed",
    [theme.breakpoints.up("sm")]: {
      position: "relative"
    }
  },
  progress: {
    backgroundColor: "transparent",
    position: "absolute",
    width: "100%",
    bottom: 0
  },
  progressBar: { background: "red" },
  youtube: {
    marginRight: theme.spacing(2),
    fontSize: "30px",
    color: "red"
  },
  title: {
    flexGrow: 1,
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block",
      color: theme.palette.grey[900]
    }
  }
});

const NavBar = ({ onSearchSubmit, classes, loading }) => {
  return (
    <AppBar elevation={1} className={classes.navBar}>
      {loading ? (
        <LinearProgress
          color="primary"
          className={classes.progress}
          classes={{
            barColorPrimary: classes.progressBar
          }}
        />
      ) : null}
      <Container>
        <Toolbar variant="regular" className={classes.toolbar}>
          <YouTubeIcon className={classes.youtube} />
          <Typography className={classes.title} variant="h6" noWrap>
            Youtube Videos
          </Typography>
          <SearchBox onSearchSubmit={onSearchSubmit} />
        </Toolbar>
      </Container>
    </AppBar>
  );
};

export default withStyles(styles)(NavBar);

NavBar.propTypes = {
  onSearchSubmit: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};
