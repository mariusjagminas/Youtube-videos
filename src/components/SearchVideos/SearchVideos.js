import React from "react";
import InputBase from "@material-ui/core/InputBase";
import SearchIcon from "@material-ui/icons/Search";
import { withStyles, fade } from "@material-ui/core/styles";
import SearchContainer from "../SearchContainer/SearchContainer";
import PropTypes from "prop-types";

const styles = theme => ({
  search: {
    position: "relative",
    color: theme.palette.grey[900],
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.5),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 1)
    },
    marginLeft: 0,
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      //TODO: what is that
      marginLeft: theme.spacing(1), //TODO: what is that
      width: "auto"
    }
  },
  searchIcon: {
    width: theme.spacing(7),
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  inputRoot: {
    color: "inherit"
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 7),
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      width: 120,
      "&:focus": {
        width: 200
      }
    }
  }
});

const SearchVideos = ({ onSearchSubmit, classes }) => {
  return (
    <SearchContainer onSearchSubmit={onSearchSubmit}>
      {(handleChange, handleSubmit, inputValue) => (
        <form onSubmit={handleSubmit} className={classes.search}>
          <div className={classes.searchIcon}>
            <SearchIcon />
          </div>
          <InputBase
            placeholder="Searchâ€¦"
            value={inputValue}
            classes={{
              root: classes.inputRoot,
              input: classes.inputInput
            }}
            onChange={handleChange}
            inputProps={{ "aria-label": "search" }}
          />
        </form>
      )}
    </SearchContainer>
  );
};

export default withStyles(styles)(SearchVideos);

SearchVideos.propTypes = {
  onSearchSubmit: PropTypes.func.isRequired
};
