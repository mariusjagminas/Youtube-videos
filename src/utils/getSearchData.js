import axios from "axios";
import { config } from "../config";

const url = "https://www.googleapis.com/youtube/v3/search";

export const getSearchData = async term => {
  try {
    const response = await axios.get(url, {
      params: {
        part: "snippet",
        maxResults: 5,
        key: config.API_KEY,
        q: term
      }
    });
    return response.data.items;
  } catch {}
};

// https://www.youtube.com/embed/ video id
