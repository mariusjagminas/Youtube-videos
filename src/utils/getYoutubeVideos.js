import axios from "axios";

const url = "https://www.googleapis.com/youtube/v3/search";

export const getYoutubeVideos = async term => {
  try {
    const response = await axios.get(url, {
      params: {
        part: "snippet",
        maxResults: 5,
        key: process.env.REACT_APP_YOUTUBE_KEY,
        q: term
      }
    });
    // excluding playlists and channels items from returned list
    const videos = await response.data.items.filter(vid => vid.id.videoId);

    return { videos: videos, error: null };
  } catch (err) {
    return { videos: [], error: err.message };
  }
};
