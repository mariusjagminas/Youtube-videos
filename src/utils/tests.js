import renderer from "react-test-renderer";
import propTypesFunc from "check-prop-types";

export const checkIfMatchSnapshot = component => {
  return it("should match a snapshot", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const checkPropTypes = (component, props) => {
  return it("should have correct propTypes", () => {
    const result = propTypesFunc(
      component.propTypes,
      props,
      "props",
      component.name
    );
    expect(result).toBe(undefined);
  });
};
