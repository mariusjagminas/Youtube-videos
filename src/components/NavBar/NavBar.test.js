import React from "react";
import { checkIfMatchSnapshot } from "../../utils/tests";
import { shallow } from "enzyme";
import NavBar from "./NavBar";
import YouTubeIcon from "@material-ui/icons/YouTube";
import SearchVideos from "../SearchVideos/SearchVideos";
import LinearProgress from "@material-ui/core/LinearProgress";

describe("<NavBar/>", () => {
  const props = {
    onSearchSubmit: jest.fn(),
    loading: false
  };
  const wrapper = shallow(<NavBar {...props} />).dive();

  it("should render the YouTube icon", () => {
    expect(wrapper.find(YouTubeIcon)).toHaveDisplayName("YouTubeIcon");
  });

  it("should render an app title", () => {
    expect(wrapper.contains("Youtube Videos")).toBeTruthy();
  });

  it("should render an input element", () => {
    expect(wrapper.find(SearchVideos)).toHaveProp("onSearchSubmit");
  });

  it("should not render the loading bar if loading={false}", () => {
    expect(wrapper.find(LinearProgress)).not.toExist();
  });

  it("should render the loading bar if loading={true}", () => {
    const wrapper = shallow(
      <NavBar onSearchSubmit={jest.fn()} loading={true} />
    ).dive();

    expect(wrapper.find(LinearProgress)).toExist();
  });

  checkIfMatchSnapshot(<NavBar {...props} />);
});
