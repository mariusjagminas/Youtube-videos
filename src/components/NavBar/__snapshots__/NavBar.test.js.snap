// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> should match a snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation1 MuiAppBar-root MuiAppBar-positionFixed NavBar-navBar-1 MuiAppBar-colorPrimary mui-fixed"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root NavBar-youtube-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
        />
      </svg>
      <h6
        className="MuiTypography-root NavBar-title-5 MuiTypography-h6 MuiTypography-noWrap"
      >
        Youtube Videos
      </h6>
      <form
        className="SearchBox-search-92"
        onSubmit={[Function]}
      >
        <div
          className="SearchBox-searchIcon-93"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="MuiInputBase-root SearchBox-inputRoot-94"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            className="MuiInputBase-input SearchBox-inputInput-95"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Searchâ€¦"
            type="text"
            value=""
          />
        </div>
      </form>
    </div>
  </div>
</header>
`;
